api/types/hardware.go

```diff
@@ -12,8 +12,53 @@ type GPU struct {
 }
 
 type HardwareSpec struct {
-	GPU     GPU                  `json:"gpu"`
-	CPU     HardwareRequestRange `json:"cpu"`
-	RAM     HardwareRequestRange `json:"ram"`
-	Storage HardwareRequestRange `json:"storage"`
+	GPU GPU                  `json:"gpu"`
+	CPU HardwareRequestRange `json:"cpu"`
+	RAM HardwareRequestRange `json:"ram"`
+	HDD HardwareRequestRange `json:"hdd"`
+}
+
+const (
+	defaultMinCPU     = 1
+	defaultMinHDD     = 4
+	defaultMinGPUs    = 1
+	defaultGPURequest = "rtx_4000"
```

Comment:
This was a mistake actually. The GPU type is provider specific so it can't go in the generic hardware spec. We probably need to validate the hardware spec in each provider.